lib/Conversion/TorchToTosa/TorchBackendLegalToTosaCustom.cpp

```diff
@@ -0,0 +1,201 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// Also available under a BSD-style license. See LICENSE.
+//
+//===----------------------------------------------------------------------===//
+
+#include "torch-mlir/Conversion/TorchToTosa/TorchToTosa.h"
+#include "torch-mlir/Conversion/TorchToTosa/TosaLegalizeCommon.h"
+#include "torch-mlir/Conversion/TorchToTosa/TosaLegalizeUtils.h"
+
+#include "../PassDetail.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+#include "mlir/Dialect/Traits.h"
+#include "mlir/IR/Matchers.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchOps.h"
+#include "torch-mlir/Dialect/Torch/Utils/Utils.h"
+#include "torch-mlir/Dialect/TorchConversion/IR/TorchConversionDialect.h"
+#include "torch-mlir/Dialect/TorchConversion/Transforms/BackendTypeConversion.h"
+#include "llvm/ADT/TypeSwitch.h"
+
+using namespace mlir;
+using namespace mlir::torch;
+using namespace mlir::torch::Torch;
+
+namespace {
+
+class ConvertBackendLegalAtenOp : public ConversionPattern {
+public:
+  ArrayRef<std::string> customOps;
+
+  ConvertBackendLegalAtenOp(TypeConverter &typeConverter, MLIRContext *context,
+                            ArrayRef<std::string> customOps)
+      : ConversionPattern(typeConverter, MatchAnyOpTypeTag{}, 0, context) {
+    this->customOps = customOps;
+  }
+
+  Value matchCustomOpOperandValue(Value inputOperand, PatternRewriter &rewriter,
+                                  Operation *backendLegalOp) const {
+    // Get the Torch Op to find the constant attributes attached to the
+    // backendLegalOp
+    Value torchInputOperand = inputOperand;
+    if (auto unrealizedCastOp = dyn_cast_or_null<UnrealizedConversionCastOp>(
+            inputOperand.getDefiningOp())) {
+      torchInputOperand = unrealizedCastOp.getInputs()[0];
+    }
+    // Handle the special case where input operand is an argument to the module
+    // function
```

Comment:
How about we add TODO here
// TODO supports more types: torch.constant.device, torch.constant.number, torch.constant.str, etc.

Reply:
Done, thanks.